pkg/proxy/utils.go

```diff
@@ -40,11 +85,40 @@ func GetSchema(ctx context.Context, c client.Client, namespace string, name stri
 	return schema, nil
 }
 
-// GetSampleDataset generates a sample dataset based on the provided dataset name.
-func GetSampleDataset(ctx context.Context, c client.Client, namespace string, datasetName string) (string, *bytes.Buffer, error) {
+// AddFileToTar adds a file with the given content to a tar.Writer.
+func AddFileToTar(tw *tar.Writer, name string, content *[]byte) error {
+	header := &tar.Header{
+		Name: name,
+		Mode: 0600,
```

Comment:
Why use `0600` file permission?

Reply:
Fixed. The permission `0600` is inherited from the initial version. However, it makes more sense to use the Linux default permission for files `0644`.